src/components/QuizGeneratorModal.tsx

```diff
@@ -120,198 +120,159 @@ export default function QuizGeneratorModal(): JSX.Element {
-                    <div className={'flex-grow'}>
-                      <label htmlFor={`question-${idx + 1}`}>
-                        Question {idx + 1}
-                      </label>
-                      <textarea
-                        id={`question-${idx + 1}`}
-                        value={question.question}
-                        onChange={e => changeQuestion(e.target.value, idx)}
-                        className={
-                          'mt-2 w-full shadow-sm focus:ring-blue-500 focus:border-blue-500 block text-sm border-gray-300 rounded-md dark:bg-gray-900 dark:border-gray-700'
-                        }
-                      />
-                      <div className={'ml-4'}>
-                        <h5 className={'font-medium my-2'}>Answers</h5>
-                        <div className={'space-y-4'}>
-                          {question.answers.map((answer, i) => (
-                            <div key={i} className={'flex'}>
-                              <div className={'mr-1'}>
-                                <button onClick={() => deleteAnswer(idx, i)}>
-                                  <XIcon className={'h-4 w-4'} />
-                                </button>
-                              </div>
-                              <div className={'w-full'}>
-                                <div
-                                  className={'w-full flex gap-6 items-center'}
-                                >
-                                  <textarea
-                                    value={answer.answer}
-                                    onChange={e =>
-                                      changeAnswer(
-                                        e.target.value,
-                                        answer.explanation,
-                                        answer.correct,
-                                        idx,
-                                        i
-                                      )
-                                    }
+    <Modal isOpen={open} onClose={closeModal}>
+      <Dialog.Panel className="px-4 text-center w-full max-w-4xl h-auto">
+        <div className="inline-block w-full max-w-4xl p-8 my-8 overflow-hidden text-left align-middle transition-all transform bg-white dark:bg-black shadow-xl rounded-2xl">
+          <div className={'flex justify-between items-center'}>
+            <Dialog.Title
+              as="h3"
+              className="text-lg font-medium leading-6 text-gray-900 dark:text-gray-200"
+            >
+              Quiz Generator
+            </Dialog.Title>
+            <button className={'btn'} onClick={() => addQuestion()}>
+              Add question
+            </button>
+          </div>
+          <div className={'space-y-8 mt-4'}>
+            {quiz.map((question, idx) => (
+              <div
+                key={idx}
+                className="flex border-l-4 focus-within:border-blue-500 dark:border-gray-800 dark:focus-within:border-blue-900"
```

Comment:
A key of idx is kind of sketchy (I think there may be some issues if you try deleting a quiz question in the middle of the list). The key should be unique per-question; one way to do this is to add a unique key to every question when you create it (ex. in `addQuestion()`, add an `id` property to the question that's `Math.random()` or `Date.now()`, and make that your key)